interval-timer.js

```diff
@@ -0,0 +1,88 @@
+import { Timer } from "./timer.js";
+import { is_function } from "./util/type-check.js";
+
+class IntervalTime extends Timer {
+  _completed_intervals = [];
+  last_completed_elapsed_time = 0;
+  constructor(
+    on_complete_interval,
+    on_start,
+    on_end,
+    on_update,
+    update_interval_rate
+  ) {
+    super(on_start, on_end, on_update, update_interval_rate);
+    this.on_complete_interval = on_complete_interval;
+  }
+
+  get completed_intervals() {
+    return this._completed_intervals;
+  }
+
+  set completed_intervals(value) {
+    this._completed_intervals.push(value);
+  }
+
+  get completed_interval() {
+    if(!this.elapsed_time){
+      throw Error("Start Timer First");
+    }
+    return this.elapsed_time - this.last_completed_elapsed_time;
+  }
+
+  update_completed_intervals() {
+    this.completed_intervals = this.completed_interval;
+  }
+
+  update_last_completed_elapsed_time() {
+    this.last_completed_elapsed_time = this.elapsed_time;
+  }
+
+  invoke_on_complete_interval() {
+    if (is_function(this.on_complete_interval)) {
+      this.on_complete_interval();
+    }
+  }
+  
+  complete_interval() {
```

Comment:
I think we should do the check for elapsed time here, otherwise we'll run into some weird problems. We should also be checking for is_running and if it's paused.

Reply:
Null check for variable elapsed_time and is_running added to function complete_interval before invoking.